src/plugins/intel_cpu/src/nodes/rdft.cpp

```diff
@@ -0,0 +1,1537 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include <vector>
+#include <cmath>
+#include <dnnl_extension_utils.h>
+#include <onednn/dnnl.h>
+#include <cpu/x64/cpu_isa_traits.hpp>
+#include <cpu/x64/jit_generator.hpp>
+#include <common/primitive_hashing_utils.hpp>
+
+#include "rdft.h"
+#include "ie_parallel.hpp"
+#include "ie_precision.hpp"
+
+#include "utils/general_utils.h"
+#include "common/cpu_memcpy.h"
+#include <openvino/op/constant.hpp>
+#include <openvino/op/rdft.hpp>
+#include <openvino/op/irdft.hpp>
+#include <sys/types.h>
+
+using namespace dnnl;
+using namespace InferenceEngine;
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu::x64;
+using namespace dnnl::impl::utils;
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+
+static const size_t DATA_INDEX = 0;
+static const size_t AXES_INDEX = 1;
+static const size_t SIGNAL_SIZE_INDEX = 2;
+
+
+bool RDFT::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (isDynamicNgraphNode(op)) {
+            errorMessage = "Doesn't support op with dynamic shapes";
+            return false;
+        }
+        const bool is_rdft = is_type<const ov::op::v9::RDFT>(op);
+        const bool is_irdft = is_type<const ov::op::v9::IRDFT>(op);
+
+        if (!is_rdft && !is_irdft) {
+            errorMessage = "Only opset9 RDFT/IRDFT operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+static void normalize_axes(std::vector<int>& axes, size_t rank, bool inverse) {
```

Comment:
Signal size input must be taken into account here as well if it exists.

Reply:
actually the sorting was unnecessary